### ISSUE TITLE
NopProvider returns true for `HasValue()`

### ISSUE THREAD

**Issue by glibsm** [uber-go/config#73]
The following test fails, yet it should succeed since NopProvider by definition can not have values.

```
package config

import (
  "testing"

  "github.com/stretchr/testify/require"
)

func TestNopProviderHasValue(t *testing.T) {
  p := NopProvider{}
  val := p.Get("some.obviously.missing.value")
  require.False(t, val.HasValue(), "nop provider can't have values")
}
```

**Comment by glibsm** [comment]
There is an existing test in the codebase that actually explicitly tests for `HasValue()` to be `true`:

```go
func TestNopProvider_Get(t *testing.T) {
	t.Parallel()

	p := NopProvider{}
	assert.Equal(t, "NopProvider", p.Name())

	v := p.Get("randomKey")
	assert.Equal(t, "NopProvider", v.Source())
	assert.True(t, v.HasValue())
	assert.Nil(t, v.Value())
}
```

This is very surprising behavior to me. Is that intentional?

**Comment by akshayjshah** [comment]
Fixed in v1.2.0.
